src/main/java/com/github/manolo8/darkbot/core/manager/PetManager.java

```diff
@@ -427,6 +432,11 @@ public void update() {
             this.fuzzyName = Strings.fuzzyMatcher(name);
             this.check = API.readMemoryLong(address, 208, 152, 0x10);
         }
+
+        @Override
+        public int hashCode() {
+            return name.hashCode();
```

Comment:
pretty sure that while you're doing this intentionally, it is not correct
The hashcode should be using all the fields, and probably the most important one, is id

I'm not totally sure, but what does ID represent for npcs? is it a random number or does the gear id match the npc id or something else?